lib/utils/fs_unix.go

```diff
@@ -19,7 +19,20 @@ limitations under the License.
 
 package utils
 
+import (
+	"os"
+	"syscall"
+)
+
 // On non-windows we just lock the target file itself.
 func getPlatformLockFilePath(path string) string {
 	return path
 }
+
+func getHardLinkCount(fi os.FileInfo) (bool, uint64) {
```

Comment:
nit: It's more usual to place the boolean return last:

```suggestion
func getHardLinkCount(fi os.FileInfo) (uint64, bool) {
```

In this case, I suppose returning just `0` or `-1` (without a bool) could work too. Up to you.

Reply:
I changed the order.  The original order made sense to me based off how they should be evaluated, but no preference really.

I think it makes sense to return a bool for two reasons:
* I want to return a uint to match the `Stat_T` typing and avoid overflow risks
* Technically having a zero hardlink count is a valid condition if the file handle does not exist on disk (for example an open file handle that has been deleted)

Reply:
I'm happy with the changed order, thanks.